charts/hivemq-operator/Chart.lock

```diff
@@ -1,6 +1,6 @@
 dependencies:
 - name: kube-prometheus-stack
   repository: https://prometheus-community.github.io/helm-charts
-  version: 30.2.0
```

Comment:
Are there any possible version incompatabilites or upgrade issues between version 30 and 61? Perhaps a serviceagent config change?

Reply:
Yeah, there is actually quite a few changes which require manual intervention. I tried to update a running legacy operator to use this new Prometheus version and it fails (expected as per the [migration guide](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md) for Prometheus):

```
Error: UPGRADE FAILED: cannot patch "operator-prometheus-node-exporter" with kind DaemonSet: DaemonSet.apps "operator-prometheus-node-exporter" is invalid: spec.selector: 
Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/instance":"operator", "app.kubernetes.io/name":"prometheus-node-exporter"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && cannot patch "operator-kube-prometheus-s-node-exporter" with kind PrometheusRule: 
Internal error occurred: failed calling webhook "prometheusrulemutate.monitoring.coreos.com": failed to call webhook: Post "https://operator-kube-prometheus-s-operator.default.svc:443/admission-prometheusrules/validate?timeout=10s": dial tcp 10.111.198.28:443: connect: connection refused
```

Also the Prometheus CRDs should be manually updated as well (Helm only create new ones but do not update existing ones)